### PR TITLE
Feat(eos_cli_config_gen): Add support for raw `eos_cli` under  `router_bgp`

### LINKED ISSUE #3570: Expand eos_cli support for BGP configuration
### Enhancement summary

Hi,

When trying to configure some features that are still missing in BGP, i believe we do not have the option of using raw_cli in some sections of the BGP configuration.

It wold be useful to be able to render raw cli commands directly under the root of BGP and the "address_family_evpn" for features that are yet to be natively supported.

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

For example, this feature (tracked support under a separate issue):
https://www.arista.com/en/support/toi/eos-4-29-0f/16349-evpn-vxlan-all-active-multihoming-gateway

Requires to append some additional BGP configuration as follows:
```
   address-family evpn
      neighbor WAN activate
      neighbor WAN domain remote
      neighbor SPINE activate
      ...
      !
      **evpn ethernet-segment domain all
         identifier 0011:1111:1111:1111:1111
         route-target import 00:01:00:01:00:01
            !
            layer-2 fec in-place update**
```
I believe eos_cli is supported under bgp vrf, vlan and vlan aware bundle only by looking at the documentation.
https://avd.arista.com/4.5/roles/eos_cli_config_gen/docs/input-variables.html#router-bgp

### Describe the solution you would like

Additional eos_cli support in BGP.

### Describe alternatives you have considered

Could not find alternatives to support this configuration yet.

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by laxmikantchintakindi**
## Change Summary

Add support for raw cli under BGP address-family-evpn.
## Related Issue(s)

Fixes #3570

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
Added feature to add raw cli to address-family-evpn in router_bgp.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml [hunk 1]
```diff
@@ -1030,6 +1030,10 @@ keys:
                 type: bool
           next_hop_unchanged:
             type: bool
+          eos_cli:
+            type: str
+            description: |
+              Multiline EOS CLI rendered directly on the Router BGP, Address family EVPN in the final EOS configuration
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml]:
```suggestion
            description: |-
              Multiline EOS CLI rendered directly on the Router BGP, Address family EVPN in the final EOS configuration.
```

**Review by ClausHolbechArista — COMMENTED** [review]
Please update the title to be more precise compared to what you are adding here.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml (hunk 1 above)
Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml]: Corrected as per suggestion.

**Comment by Shivani-gslab** [comment]
As per the issue description, do we need to add `eos_cli` under `router_bgp.bgp` or `router_bgp`as well ?
@ClausHolbechArista @gmuloc

**Comment by ClausHolbechArista** [comment]
> As per the issue description, do we need to add `eos_cli` under `router_bgp.bgp` or `router_bgp`as well ? @ClausHolbechArista @gmuloc

From the issue I read it as we should just add `router_bgp.eos_cli`. There is no need to add it specifically under `address_family_evpn` since that is a static CLI, which can just be included in the CLI the user puts under `eos_cli`.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml [hunk 2]
```diff
@@ -1030,6 +1030,10 @@ keys:
                 type: bool
           next_hop_unchanged:
             type: bool
+          eos_cli:
```
Comment (Shivani-gslab) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml]: Please remove this from address_family.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2 [hunk 3]
```diff
@@ -1741,4 +1741,8 @@ router bgp {{ router_bgp.as }}
       recovery delay {{ session_tracker.recovery_delay }} seconds
 {%         endif %}
 {%     endfor %}
+{%     if router_bgp.eos_cli is arista.avd.defined %}
+   !
+   {{ router_bgp.eos_cli | indent(6, false) }}
+{%     endif %}
```
Comment (Shivani-gslab) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2:1747]: you can add a molecule test for this feature.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml (hunk 2 above)
Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml]: Removed

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2 (hunk 3 above)
Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2:1747]: Added the test.

**Review by sugetha24 — APPROVED** [review]
LGTM.
Tested: Molecule tests with various options under `address-family evpn`

**Review by gmuloc — APPROVED** [review]

**Pull request merged**